include/openenclave/edger8r/host.h

```diff
@@ -106,6 +106,14 @@ OE_INLINE size_t oe_wcslen(const wchar_t* s)
     return wcslen(s);
 }
 
+#if defined(__GNUC__)
+#define OE_WEAK __attribute__((weak))
+#elif _MSC_VER
+#define OE_WEAK __declspec(selectany)
```

Comment:
Looks like you're right. I guess this won't work then.

I tried to prefix symbol names with the file name and then just do include guards around in inline function in the header like this

```
OE_INLINE int my_ecall() {
   return filename_prefixed_my_ecall();
}
```

But that doesn't work because when you compile the header for use in the OE core libraries, the symbol `filename_prefixed_my_ecall` will not be in the binary (it will have some other prefix with a different filename).

Reply:
Very challenging problem. I just wish `selectany` would work.
@mingweishih Any ideas?

Reply:
It may work based on the [source](https://github.com/microsoft/cppwinrt/blob/4d8e0e26ad9c8728560bf62840bce4f5364d8779/strings/base_extern.h#L2). Maybe worth verifying?

Reply:
Well, it seems a function pointer 😢

Reply:
We could prefix the functions and then declare a function pointer selectany. So we would generate the following for each ecall:

```
int filename_prefixed_my_ecall() {
   // do stuff
}

__declspec(selectany) int (*my_ecall)() = filename_prefixed_my_ecall;
```

Reply:
Sounds like a plan. With function pointers, do we still need prefixing with filename?

Reply:
On another note, I'm fine with this going in just for Linux and another PR to tackle the windows specific problem.

The only outstanding items seems to be "weak symbols for windows"; and I think that can be tackled in another PR, if it makes development easy.

Reply:
The oeedger8r test fails if this isn't addressed, so I think I'll have to address this to merge this PR

Reply:
Ok I have implemented this here: https://github.com/openenclave/openenclave/pull/2961/commits/3ed98e3cff51fc11d708c59f57fd1cc7ae118091